src/Widgets/Page.vala

```diff
@@ -108,13 +113,27 @@ namespace Network.Widgets {
                 try {
                     device.disconnect (null);
                 } catch (Error e) {
-                    warning (e.message);
+                    show_dialog (
+                        _("Failed To Disconnect"),
+                        _("Could not disconnect: %s.").printf (e.message),
+                        "dialog-error",
+                        Gtk.ButtonsType.CLOSE
+                    );
+
+                    control_switch.active = true;
                 }
             } else if (control_switch.active && device.get_state () == NM.DeviceState.DISCONNECTED) {
                 var connection = NM.SimpleConnection.new ();
                 var remote_array = device.get_available_connections ();
                 if (remote_array == null) {
-                    this.show_error ();
+                    show_dialog (
+                        _("Failed To Connect"),
```

Comment:
Can we be a little more specific here? What did we fail to connect to?

Reply:
We failed to connect with the current connection, if you have a connection that's setup incorrectly, that is, DNS is wrong or anything else, this will show.

Reply:
So then would it make sense to show something like `Failed to connect to "SSID Name"`?